sql/core/src/test/scala/org/apache/spark/sql/execution/ExternalAppendOnlyUnsafeRowArrayBenchmark.scala

```diff
@@ -20,13 +20,27 @@ package org.apache.spark.sql.execution
 import scala.collection.mutable.ArrayBuffer
 
 import org.apache.spark.{SparkConf, SparkContext, SparkEnv, TaskContext}
-import org.apache.spark.benchmark.Benchmark
+import org.apache.spark.benchmark.{Benchmark, BenchmarkBase}
 import org.apache.spark.internal.config
 import org.apache.spark.memory.MemoryTestingUtils
 import org.apache.spark.sql.catalyst.expressions.UnsafeRow
 import org.apache.spark.util.collection.unsafe.sort.UnsafeExternalSorter
 
-object ExternalAppendOnlyUnsafeRowArrayBenchmark {
+/**
+  * Synthetic Benchmark for ArrayBuffer and UnsafeRowArray
+  * To run this benchmark:
+  * {{{
+  *   1. without sbt:
+  *      bin/spark-submit --class <this class> --jars <spark core test jar> <spark catalyst test jar>
+  *   2. build/sbt "catalyst/test:runMain <this class>"
+  *   3. generate result:
+  *      SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "catalyst/test:runMain <this class>"
+  *      Results will be written to
+  *      "benchmarks/ExternalAppendOnlyUnsafeRowArrayBenchmark-results.txt".
+  * }}}
+  */
+
+object ExternalAppendOnlyUnsafeRowArrayBenchmark extends BenchmarkBase {
```

Comment:
Sorry @heary-cao Seems it's duplicate work? https://github.com/apache/spark/pull/22617

Reply:
Please check our all refactors here: https://issues.apache.org/jira/browse/SPARK-25475